src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -4679,6 +4679,11 @@ void TestParameterListStruct([AsParameters] ParameterListStruct args)
                 args.HttpContext.Items.Add("input", args.Value);
             }
 
+            void TestParameterListNullableStruct([AsParameters] ParameterListStruct? args)
```

Comment:
Add a test that doesn't pass any bindable values and check that the struct is null?

Reply:
I don't know if it will ever be null based on my change.

- If any bindable values is passed, and they are required, it should cause a `BadRequestException` (or Bad Request response).

- When the properties are not required it should create an instance and have the default values for all properties.

What do you think about this?

Reply:
> What do you think about this?

That makes sense to me. 

> When the properties are not required it should create an instance and have the default values for all properties.

Is there a test for this?

Reply:
> Is there a test for this?

Just added one.

Reply:
Are we sure about this? This feels like the opposite direction we went with nullable `StringValues`. See the conversation at https://github.com/dotnet/aspnetcore/pull/41789#issuecomment-1181263769

We decided that it should be impossible to ever get a `default(StringValues)` as an argument to either a `StringValues` or a `StringValues?` parameter. If I declare a parameter to be nullable, I expect to be given a `null` nullable when nothing has been provided.

Are we doing the same thing for optional non-struct parameter? Are we just filling each missing property with its default value if it's marked nullable and the property is not?

What exactly happens when only a subset of required properties are provided for a nullable reference type parameter? What about a nullable struct?

Reply:
I got your point about the this be a opposite direction; however, I see the `AsParameters` as an approach to create `structured parameters` where we are just organizing the original parameters in a `class`/`struct`. The actual binding is exactly the same on the properties (all default values and optionality), including the BadRequest when they are missing, and the actual `AsParameters` parameter does not contribute to the optionality (setting it as nullable will not make all properties optional).

> Are we doing the same thing for optional non-struct parameter? 

We do the same thing but as I mention, today, a nullable `AsParameters` parameter does not mean `optional`. If we can bound all properties (default value, null value or actual value from a source) an instance of the `AsParameters` parameter will be created

> What exactly happens when only a subset of required properties are provided for a nullable reference type parameter? What about a nullable struct?

The `AsParameters` parameter does not change the behavior of the binding. If the request misses a required property a Bad Request exception/response will be produced.

In general, a nullable `AsParameters` parameter should not exists :) and maybe that could be the right option, but I don't see a strong reason to throw if a nullable is defined and I prefer support it, but I would love to hear other opinions.

Reply:
> and the actual `AsParameters` parameter does not contribute to the optionality (setting it as nullable will not make all properties optional).

Can we just throw if the `AsParameters` parameter is marked nullable since we ignore this and always pass a value anyway?

Reply:
Yes we could.

>In general, a nullable AsParameters parameter should not exists :) and maybe that could be the right option, but I don't see a strong reason to throw if a nullable is defined and I prefer support it, but I would love to hear other opinions.

Reply:
That's how I feel, nullable can be annoying to deal with in user code, plus I might expect to get a null `[AsParameters]` param value if not all values were provided by the user (which isn't what would happen). So throwing at startup would get rid of any ambiguity.

Reply:
Based what I am understanding, the preference is to throw for nullable `class`/`struct` with `AsParameters`. Is that correct? If so, I will update the PR with that.

Reply:
> Is that correct?

I think so. That's definitely my position.

Reply:
Yep, I agree with throwing an exception here as Bruno stated:

> In general, a nullable AsParameters parameter should not exists 

The explicit behavior of throwing an exception seems better to me here than the implicit behavior of handling setting default values for a nullable struct with AsParameters.

Reply:
PR changed to throw when nullable.